spec/support/api_spec_helper.rb

```diff
@@ -6,47 +6,32 @@
 require 'json'
 
 module ApiSpecHelper
-  HEADER_ALIASES = {
-    "auth_token" => "HTTP_X_AUTH_TOKEN",
-    "miq_token"  => "HTTP_X_MIQ_TOKEN",
-    "miq_group"  => "HTTP_X_MIQ_GROUP"
-  }
-
-  DEF_HEADERS = {
-    "Content-Type" => "application/json",
-    "Accept"       => "application/json"
-  }
-
-  def update_headers(headers)
-    HEADER_ALIASES.keys.each do |k|
-      if headers.key?(k)
-        headers[HEADER_ALIASES[k]] = headers[k]
-        headers.delete(k)
-      end
-    end
-    headers.merge!("HTTP_AUTHORIZATION" => @http_authorization) if @http_authorization
-    headers.merge(DEF_HEADERS)
+  def request_headers
+    @request_headers ||= {
+      "Content-Type" => "application/json",
+      "Accept"       => "application/json"
+    }
```

Comment:
@Fryguy that was intentional, because I want `request_headers` to be mutable to do things like https://github.com/ManageIQ/manageiq/pull/11258/files#diff-984450d5f2a8800084e77dcc0a785b2aR99. I never liked what was there before, i.e. having to set an ivar so that it could be added to the headers later, if it was set.

I find it odd that Rails' integration specs has one method for setting the `"Accept"` header prior to request time, but that's it and there's no support for setting whatever headers you want, as you can do with controller tests. It would be cool if support for that could be added.....while that's not there I'm trying to add an API somewhat analogous to the controller specs where you can do `env.request["Content-Type"] = "application/json"`, etc...